analysis_driver/app_logging.py

```diff
@@ -1,57 +1,109 @@
-__author__ = 'mwham'
 import logging
-from analysis_driver.config import logging_default as log_cfg
+import logging.config
+import logging.handlers
+from analysis_driver.config import default as cfg
 
 
-class AppLogger:
-    """
-    Mixin class for logging. An object subclassing this can log directly from itself. Contains a
-    logging.Logger object, which is controlled by public methods. All methods are voids, i.e. return None.
-    """
-    _logger = None
+class LoggingConfiguration:
+    """Stores Loggers, Formatters and Handlers"""
+    def __init__(self):
+        self.default_formatter = logging.Formatter(
+            fmt=cfg.query('logging', 'format', ret_default='[%(asctime)s][%(name)s][%(levelname)s] %(message)s'),
+            datefmt=cfg.query('logging', 'datefmt', ret_default='%Y-%b-%d %H:%M:%S')
+        )
+        self.blank_formatter = logging.Formatter()
+        self.formatter = self.default_formatter
+        self.handlers = set()
+        self.loggers = set()
+        self.log_level = logging.INFO
+
+    def get_logger(self, name, level=None):
+        """
```

Comment:
Would we ever need to set up a logger without registering it to all the handlers? (e.g. clarity, rest_communication)